tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,321 @@
-
-        // Try to lock the session file to ensure it's unlocked
-        const fd = fs.openSync(sessionFile, 'r');
-        expect(lock(fd)).toBeTruthy();
-        lock.unlock(fd);
-        fs.closeSync(fd);
+    // Run command that will fail
+    const { exitCode } = await testBinUtils.pkStdio(
+      ['identities', 'search', 'InvalidProvider'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
       },
-      global.failedConnectionTimeout,
+      global.binAgentDir,
     );
-
-    test('Parallel processes should not refresh the session token', async () => {
-      let tokenP1 = 'token1' as SessionToken;
-      let tokenP2 = 'token2' as SessionToken;
-
-      tokenBuffer = await fs.promises.readFile(sessionFile);
-      const prevTokenParallel = tokenBuffer.toString() as SessionToken;
-
-      async function runListCommand(): Promise<SessionToken> {
-        // At least 1 second of delay
-        // Expiry time resolution is in seconds
-        await sleep(1000);
-        await testUtils.pkStdio(command, {}, dataDir);
-        const buffer = await fs.promises.readFile(sessionFile);
-        return buffer.toString() as SessionToken;
-      }
-
-      [tokenP1, tokenP2] = await Promise.all([
-        runListCommand(),
-        runListCommand(),
-      ]);
-
-      // Verify both tokens
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(tokenP1),
-      ).not.toThrow();
-      expect(
-        async () => await polykeyAgent.sessionManager.verifyToken(tokenP2),
-      ).not.toThrow();
-
-      // Check that both commands were completed
-      expect(tokenP1).not.toEqual('token1');
-      expect(tokenP2).not.toEqual('token2');
-
-      // Check that the session token was refreshed exactly one time
-      if (tokenP1 === prevTokenParallel) {
-        expect(tokenP2).not.toEqual(prevTokenParallel);
-      } else if (tokenP2 === prevTokenParallel) {
-        expect(tokenP1).not.toEqual(prevTokenParallel);
-      } else {
-        expect(tokenP1).toEqual(tokenP2);
-      }
+    expect(exitCode).not.toBe(0);
+    await sleep(1100);
+    // Write to session file to ensure it's unlocked
+    await session.writeToken('abc' as SessionToken);
+    const token = await session.readToken();
+    expect(token).toEqual('abc');
+  });
+  test('agent lock should remove the token from the client and delete the token', async () => {
+    const { exitCode } = await testBinUtils.pkStdio(
+      ['agent', 'lock'],
+      {
+        PK_NODE_PATH: global.binAgentDir,
+        PK_PASSWORD: global.binAgentPassword,
+      },
+      global.binAgentDir,
+    );
+    expect(exitCode).toBe(0);
+    await expect(
+      fs.promises.readdir(path.join(global.binAgentDir)),
+    ).resolves.not.toContain('token');
+  });
+  test('cause old sessions to fail when locking all sessions', async () => {
+    // Generate new token
+    await testBinUtils.pkStdio(
+      ['agent', 'unlock'],
```

Comment:
You might as well add in a test for `lockall` too then at this point.

Later we can refactor some locking tests to be in `lock.test.ts`and `unlock.test.ts`and `lockall.test.ts`. Then `sessions.test.ts` will be what is left over.

Reply:
Specifically I wanted `sessions.test.ts` to be precisely just the CARL loop.

Reply:
All of the locking tests have been moved to new files in `bin/agent/unlock.test.ts`, `bin/agent/lock.test.ts`, and `bin/agent/lockall.test.ts`.